modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/function/VolatilityFunctionProvider.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.impl.volatility.smile.function;
+
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.math.impl.function.Function1D;
+import com.opengamma.strata.pricer.impl.option.EuropeanVanillaOption;
+
+/**
+ * Provides functions that return volatility and its sensitivity to volatility model parameters. 
+ * 
+ * @param <T> Type of the data needed for the volatility function
+ */
+public abstract class VolatilityFunctionProvider<T extends SmileModelData> {
+
+  private static final double EPS = 1.0e-6;
+
+  /**
+   * Computes the volatility. 
+   * 
+   * @param option  the option
+   * @param forward  the forward value of the underlying
+   * @param data  the model data
+   * @return the volatility
+   */
+  public abstract double getVolatility(EuropeanVanillaOption option, double forward, T data);
+
+  /**
+   * Computes the volatility sensitivity to the model parameters by means of central finite difference. 
```

Comment:
Should indicate that a default central FD is implemented at this level but is likely overridden in the non-abstract implementations.